### ISSUE TITLE
memory.js-style "before" middleware does not appear to work without next() call

### ISSUE THREAD

**Issue by sergebat** [englercj/resource-loader#29]
There's an example for implementation of "caching" before middleware here:
https://github.com/englercj/resource-loader/blob/master/src/middlewares/caching/memory.js

```
        if (cache[resource.url]) {
            resource.data = cache[resource.url];
            resource.complete();
        }
        // if not cached, wait for complete and store it in the cache.
        else {
            resource.once('complete', function () {
               cache[this.url] = this.data;
            });

            next();
        }
```

Example does not call next(), if resource is cached and calls complete() on the resource instead. 

However, it really seems that in that such case _boundOnLoad is never called and thus progress is stuck:
https://github.com/englercj/resource-loader/blob/master/src/Loader.js#L388 

Am I missing something, or memory.js is not working as expected? 

I was hoping to use the same approach to implement totally custom sound loading with howlerjs, like that:
```
        if (resource.loadType === PIXI.loaders.Resource.LOAD_TYPE.AUDIO) {
            var data = new Howl({
                urls: [resource.url],
                autoplay: false,
                onload: function() {
                    resource.data = data;
                    sounds[resource.name] = resource.data;
                    resource.complete();
                }
            });
        }
        else {
            next();
        }
```

**Comment by englercj** [comment]
I wrote memory.js a while ago before the library evolved a bit, it will likely need tweaking to work properly. I will make sure to look at this soon, thanks!

**Comment by englercj** [comment]
Fixed by #34.
